src/nvim/api/buffer.c

```diff
@@ -411,10 +411,13 @@ void nvim_buf_set_lines(uint64_t channel_id,
 
 /// Gets a buffer-scoped (b:) variable.
 ///
-/// @param buffer     Buffer handle
-/// @param name       Variable name
-/// @param[out] err   Error details, if any
-/// @return Variable value
+/// @note Use buffer_get_changedtick() for “variable” `b:changedtick`.
+///       This method only gets variables from b: dictionary.
```

Comment:
@justinmk Current behaviour is consistent with `b:['changedtick']`, so I do not want to make this an error.